dev/js/authenticate.js

```diff
@@ -1,23 +1,34 @@
-var startApp = function() {
+let startApp = function() {
+	let startAppHttpCommunicator = new httpCommunicator();
+	let startAppHtmlElementManager = new htmlElementStyleManager(document.getElementById('gSignInWrapper'),	document.getElementById("gSignOutWrapper"));
 	gapi.load('auth2', function(){
-		var startAppHttpCommunicator = new httpCommunicator();
-		var startAppHtmlElementManager = new htmlElementStyleManager(document.getElementById('gSignInWrapper'),
-			document.getElementById("gSignOutWrapper"));
-		// Retrieve the singleton for the GoogleAuth library and set up the client.
 		auth2 = gapi.auth2.init({
 			client_id: '856845744679-fr7uheupsjm65udbao75b6no8vjl8cm0.apps.googleusercontent.com',
 			cookiepolicy: 'single_host_origin',
 			scope: 'https://www.googleapis.com/auth/calendar'
 		});
-	
 		auth2.attachClickHandler(
 			document.getElementById('customBtn'), //what document element is the click attached to
 			{}, //OPTIONS
 			function() {auth2.grantOfflineAccess().then(function(resp) {
 				console.log(resp.code);
 				startAppHtmlElementManager.swapVisibility(0,1,"hidden","visible");
-				var dataToBeSent = {auth_code: resp.code}
-				startAppHttpCommunicator.sendToServer("http://127.0.0.1:5000/login",dataToBeSent);
+				let dataToBeSent = {auth_code: resp.code}
+				startAppHttpCommunicator.communicateWithServer("http://127.0.0.1:5000/login",dataToBeSent,"POST").then(function(resp){
+					if(resp == 1){
```

Comment:
Maybe use trim, if it's a problem with new line.

https://www.w3schools.com/jsref/jsref_trim_string.asp